jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ServerFilter.java

```diff
@@ -85,6 +85,8 @@ public void filter(ContainerRequestContext containerRequestContext) throws IOExc
       }
       Span serverSpan = builder.startManual();
 
+      tracer.makeActive(serverSpan);
+
       traceContext.push(serverSpan);
```

Comment:
I don't think using both the traceContext and active span is a good idea, they can easily diverge. I thought we would reimplement the trace context via the active span API.

Reply:
Oh, I misunderstood. I had made an assumption that the ActiveSpanSource used Jaeger's ThreadLocal storage implementation, but I see that this is not the case. 

I feel that having Jaeger's `ThreadLocalTraceContext` implement `io.opentracing.ActiveSpanSource` might work better for interoperability with code using older instrumentation. I think that we should switch to using `io.opentracing.util.ThreadLocalActiveSpanSource` when we update instrumentation. 

Is this what you were proposing?

Reply:
I was thinking `ThreadLocalTraceContext` would delegate to `ActiveSpanSource`, but implementing it might be even better, although I would still prefer that implementation delegates to the existing ref-counting OT implementation of the `ActiveSpanSource`. The main trick is that currently `ThreadLocalTraceContext` is independent of the tracer, we may need to make a breaking API change to force them to be bound together.

Reply:
Sounds reasonable, I'll attempt to reuse what we can, and put my work up in a separate commit. (Will update this once that is merged)

I haven't thought through the API changes yet, but I'm hoping that it ends up being non breaking for the users who only access the `ThreadLocalTraceContext` statically through `TracingUtils`.

Reply:
> who only access the ThreadLocalTraceContext statically through TracingUtils.

+1